pkg/controller/provisioner.go

```diff
@@ -24,15 +24,22 @@ var (
 	}
 )
 
-// Extract available SAS addresses for the Node from topology segments
-func parseTopology(topology []*csi.Topology, parameters *map[string]string) error {
-	klog.Infof("Topology: %v", topology)
-
-	sasAddressSearchString := fmt.Sprintf("%s/%s", common.TopologyInitiatorPrefix, common.TopologySASInitiatorLabel)
-	for _, topo := range topology {
-		for key, val := range topo.GetSegments() {
+// Extract available SAS addresses for Nodes from topology segments
+// This will contain all SAS initiators for all nodes unless the storage class
+// has specified allowed or preferred topologies
+func parseTopology(topologies []*csi.Topology, parameters *map[string]string) error {
```

Comment:
Does this get called for iSCSI and FC?

Reply:
You're right this could have been moved inside a conditional based on storage protocol, however there is a upcoming change for node verification in controllerpublishvolume that will need some of the information from the topology map for all protocols